ts/tests/commands/precompile-test.ts

```diff
@@ -79,4 +79,21 @@ describe('Acceptance: ts:precompile command', function() {
         });
     });
   });
+
+  it('generates .d.ts files when addon and package names do not match', function() {
+    fs.ensureDirSync('addon-test-support');
+    fs.writeFileSync('addon-test-support/test-file.ts', `export const testString: string = 'hello';`);
+    fs.writeFileSync('index.js', `module.exports = { name: 'my-addon' };`);
+
+    const pkg = fs.readJSONSync('package.json');
+    pkg.name = '@foo/my-addon';  // addon `name` is `my-addon`
+    fs.writeJSONSync('package.json', pkg);
+
+    return ember(['ts:precompile'])
+      .then(() => {
```

Comment:
as of #644 you could just pass an async function to `it` and make this an `await`

Reply:
@mike-north oh, nice. I'd be inclined to do that and a few other things in a follow-on modernization pass – e.g. I also want to add and use Prettier.